src/lib/hooks/routing/useClientSideSmartOrderRouterTrade.ts

```diff
@@ -0,0 +1,122 @@
+import { Currency, CurrencyAmount, TradeType } from '@uniswap/sdk-core'
+import { useStablecoinAmountFromFiatValue } from 'hooks/useUSDCPrice'
+import { useEffect, useMemo, useState } from 'react'
+import { GetQuoteResult, InterfaceTrade, TradeState } from 'state/routing/types'
+import { computeRoutes, transformRoutesToTrade } from 'state/routing/utils'
+
+import { getClientSideQuote } from './clientSideSmartOrderRouter'
+import { useRoutingAPIArguments } from './useRoutingAPIArguments'
+
+export default function useClientSideSmartOrderRouterTrade<TTradeType extends TradeType>(
+  tradeType: TTradeType,
+  amountSpecified?: CurrencyAmount<Currency>,
+  otherCurrency?: Currency
+): {
+  state: TradeState
+  trade: InterfaceTrade<Currency, Currency, TTradeType> | undefined
+} {
+  const [currencyIn, currencyOut]: [Currency | undefined, Currency | undefined] = useMemo(
+    () =>
+      tradeType === TradeType.EXACT_INPUT
+        ? [amountSpecified?.currency, otherCurrency]
+        : [otherCurrency, amountSpecified?.currency],
+    [amountSpecified, otherCurrency, tradeType]
+  )
+
+  const queryArgs = useRoutingAPIArguments({
+    tokenIn: currencyIn,
+    tokenOut: currencyOut,
+    amount: amountSpecified,
+    tradeType,
+    useClientSideRouter: true,
+  })
+
+  const [loading, setLoading] = useState(false)
+  const [fetchedResult, setFetchedResult] = useState<
+    | {
+        data: GetQuoteResult | undefined
+        error?: unknown
+      }
+    | undefined
+  >()
+
+  useEffect(() => {
+    async function fetchQuote() {
```

Comment:
If the only thing this depends on is `queryArgs`, move the definition outside of the component, and pass in `queryArgs` explicitly, to avoid the inline definition and long component definition.

Reply:
Would need to accept the state setters too, is that weird?